include/common/mem.h

```diff
@@ -0,0 +1,51 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+#ifndef __LABWC_MEM_H
+#define __LABWC_MEM_H
+
+#include <stdlib.h>
+
+/*
+ * As defined in busybox, weston, etc.
+ * Allocates zero-filled memory; calls exit() on error.
+ * Returns NULL only if (size == 0).
+ */
+void *xzalloc(size_t size);
+
+/*
+ * Type-safe macros in the style of C++ new/new[].
+ * <expr> may be either a type name or value expression.
+ *
+ * Examples:
+ *   struct wlr_box *box = znew(*box);
+ *   char *buf = znew_n(char, 80);
```

Comment:
another example for `struct wlr_box *box = znew(struct wlr_box);` would be nice here

Reply:
I thought our coding style was going to recommend against that style though.

Reply:
With the type check in place I personally don't really care all that much if we 'allow' both variants or just the `znew(*box)` one. We can always update this later on again once we merge #556.

So happy to drop this point for now.